### ISSUE TITLE
bug(datepicker) wrong rendering with ng-template inside mat-form-field

### ISSUE THREAD

**Issue by mattiacorci** [angular/components#15009]
#### What is the expected behavior?
I expected to get the right rendering when using a `matDatepicker` inside a `ng-template`, inside a `mat-form-field`.

#### What is the current behavior?
But currently, the size and the position of the `mat-datepicker-toggle` button are wrong. Inspecting the html, it seems like that, in this case, the component is not moving the `mat-datepicker-toggle` button inside the `mat-form-field-suffix` element.

#### What are the steps to reproduce?
https://angular-material2-issue-putyxt.stackblitz.io

https://stackblitz.com/edit/angular-material2-issue-putyxt?file=app%2Fapp.component.html

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
OS: Windows 7
Angular: Latest
Angular Material: Latest

#### Is there anything else we should know?
Why I'm doing this?
I'm trying to create a component where the form is dinamically rendered depending by a configuration.
To get this result I'm using an `ngFor` in the `mat-form-field` element and then I'm adding the specific input type inside the element.

**Comment by etay2000** [comment]
The same issue occurs when trying to use matPrefix or matSuffix inside the template of a custom MatFormFieldControl. 

```
@Component({
  selector: 'px-widget',
  template: '
    <span matPrefix>Prefix</span>
    <input matInput>
    <span matSuffix>Suffix</span>
  ',
  styles: [],
  providers: [
    {
      provide: MatFormFieldControl,
      useExisting: PxWidgetComponent
    }
  ]
})
export class PxWidgetComponent implements MatFormFieldControl<string> {

  readonly autofilled: boolean;
  readonly controlType: string;
  readonly disabled: boolean;
  readonly empty: boolean;
  readonly errorState: boolean;
  readonly focused: boolean;
  readonly id: string = 'px-widget-0';
  readonly ngControl: NgControl | null;

  @Input()
  placeholder: string;
  readonly required: boolean;
  readonly shouldLabelFloat: boolean;

  stateChanges: Subject<void> = new Subject<void>();

  value: string | null;

  constructor() {

  }

}
```

```
<!-- Usage -->
<mat-form-field>
  <px-widget></px-widget>
</mat-form-field>
```
The matPrefix and matSuffix are never recognized by the mat-form-field and therefore never properly aligned.

**Comment by Aleksandyr** [comment]
Hello @mattiacorci and @etay2000 it seems that the problem comes from the Angular framework itself.
I've done the research around this matter and it turns out that for now there is no way to access nested references to a elements (children) from a host element of the component unless they are not direct descendants.

What does this mean?
There is no way that you can detect from a **mat-form-filed** component whether there is a nested components or elements which contains **matSuffix** **matPreffix** directives or whatever it is.

You can see the feature request [#6728](https://github.com/angular/material2/issues/6728) which was proposed here. I think that once it is resolved you would be able to achieve the approach you are trying.
There is a slight showstopper which prevents or somehow slows down the implementation of this feature and this is the [feature request](https://github.com/angular/angular/issues/24858) which is logged into [Angular](https://github.com/angular/angular/) repository.

I have encountered several issues which are related and somehow similar to this problem.
You can find them here:
- [[angular/material2 [12060]]](https://github.com/angular/material2/issues/12060)
- [ngx-formly [715]](https://github.com/ngx-formly/ngx-formly/issues/715)
- [StackOverflow](https://stackoverflow.com/questions/50467206/mat-error-put-to-separate-component-not-rendering-properly/50475917#50475917)

Unfortunately I couldn't manage to find a workaround for this but if I find one I will let you know.

**Comment by jelbourn** [comment]
As @Aleksandyr said, this is unfortunately a limitation of the framework; the form-field doesn't have a way to query through a template for content children. There is special handling in Angular for when `*ngIf` or `*ngFor` are _directly_ on a content child, but it doesn't extend to using `ng-template`.

**Comment by etay2000** [comment]
So after 5 months we now know it is a limitation of the framework. Should we also abandon any hope of this being addressed by future versions 9.01, 10.13, 11.987, 12.36, 20 (or whatever the latest release is being called), etc...  like this issue in the angular/angular repo [https://github.com/angular/angular/issues/24858](url). Does the tag 'fixed by ivy' actually mean anything or is that just a way to say 'Don't get your hopes up, won't be fixing this anytime soon!"?
